xds/internal/balancer/edsbalancer/edsbalancer_test.go

```diff
@@ -364,6 +379,65 @@ func TestEDS_TwoLocalities(t *testing.T) {
 	}
 }
 
+// The EDS balancer gets EDS resp with unhealthy endpoints. Test that only
+// healthy ones are used.
+func TestEDS_EndpointsHealth(t *testing.T) {
+	cc := newTestClientConn(t)
+	edsb := NewXDSBalancer(cc, nil)
+
+	// Two localities, each 3 backend, one Healthy, one Unhealthy, one Unknown.
+	clab1 := newClusterLoadAssignmentBuilder(testClusterNames[0], nil)
+	clab1.addLocality(testSubZones[0], 1, testEndpointAddrs[:6], &addLocalityOptions{
+		health: []corepb.HealthStatus{
+			corepb.HealthStatus_HEALTHY,
+			corepb.HealthStatus_UNHEALTHY,
+			corepb.HealthStatus_UNKNOWN,
+			corepb.HealthStatus_DRAINING,
+			corepb.HealthStatus_TIMEOUT,
+			corepb.HealthStatus_DEGRADED,
+		},
+	})
+	clab1.addLocality(testSubZones[1], 1, testEndpointAddrs[6:12], &addLocalityOptions{
+		health: []corepb.HealthStatus{
+			corepb.HealthStatus_HEALTHY,
+			corepb.HealthStatus_UNHEALTHY,
+			corepb.HealthStatus_UNKNOWN,
+			corepb.HealthStatus_DRAINING,
+			corepb.HealthStatus_TIMEOUT,
+			corepb.HealthStatus_DEGRADED,
+		},
+	})
+	edsb.HandleEDSResponse(clab1.build())
+
+	var readySCs []balancer.SubConn
+	for i := 0; i < 4; i++ {
+		sc := <-cc.newSubConnCh
```

Comment:
Is there some way we can confirm these are the right backends (0/2/6/8)?  I'm moderately worried about the way this test works: expecting 4 subconns to be created and then waiting a brief period to ensure there isn't a 5th.  If we're confirming it's the proper backends then it wouldn't be predicated as much on timing.

Reply:
Done